underfs/abfs/src/main/java/alluxio/underfs/abfs/AbfsUnderFileSystem.java

```diff
@@ -66,8 +67,28 @@ private static Configuration createAbfsConfiguration(UnderFileSystemConfiguratio
             conf.getString(PropertyKey.ABFS_CLIENT_ID));
         abfsConf.set(PropertyKey.ABFS_CLIENT_SECRET.getName(),
             conf.getString(PropertyKey.ABFS_CLIENT_SECRET));
+        clientCredentials = true;
       }
     }
+
+    if (!clientCredentials && !sharedKey) {
+      abfsConf.set("fs.azure.account.auth.type", "OAuth");
+      abfsConf.set("fs.azure.account.oauth.provider.type",
+              "org.apache.hadoop.fs.azurebfs.oauth2.MsiTokenProvider<");
+      if (conf.isSet(PropertyKey.ABFS_MSI_ENDPOINT)) {
+        abfsConf.set(PropertyKey.ABFS_MSI_ENDPOINT.getName(),
+            conf.getString(PropertyKey.ABFS_MSI_ENDPOINT));
+      }
+      if (conf.isSet(PropertyKey.ABFS_MSI_TENANT)) {
+        abfsConf.set(PropertyKey.ABFS_MSI_TENANT.getName(),
+            conf.getString(PropertyKey.ABFS_MSI_TENANT));
+      }
+      if (conf.isSet(PropertyKey.ABFS_CLIENT_ID)) {
```

Comment:
do we need all of `ABFS_MSI_ENDPOINT`, `ABFS_MSI_TENANT`, `ABFS_CLIENT_ID` set in the configuration at the same time? or just part of them is ok?

Reply:
all of the parameters is options, but if it has some special aad properties, thease parameter should be set separately.